cmake/OpenCVFindLibsGrfmt.cmake

```diff
@@ -251,9 +251,25 @@ if(WITH_PNG)
   endif()
 
   set(HAVE_PNG YES)
+  set(HAVE_SPNG NO)
   set(PNG_VERSION "${PNG_LIBPNG_VER_MAJOR}.${PNG_LIBPNG_VER_MINOR}.${PNG_LIBPNG_VER_RELEASE}")
 endif()
 
+if(WITH_SPNG)
+  set(SPNG_LIBRARY libspng CACHE INTERNAL "")
+  set(SPNG_LIBRARIES ${SPNG_LIBRARY})
+  add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/libspng")
+  set(SPNG_INCLUDE_DIR "${${SPNG_LIBRARY}_SOURCE_DIR}" CACHE INTERNAL "")
+  set(SPNG_DEFINITIONS "")
+  ocv_parse_header("${SPNG_INCLUDE_DIR}/spng.h" SPNG_VERSION_LINES SPNG_VERSION_MAJOR SPNG_VERSION_MINOR SPNG_VERSION_PATCH)
+
+  set(HAVE_SPNG YES)
+  #set(HAVE_PNG NO)
```

Comment:
I propose to uncomment the line to not have 2 codecs at a time.

Reply:
This was intended to surpress CMake warning while running the workflow. I made libspng default png codec to test in the workflow. Before merge, i will uncomment that line and set `OCV_OPTION(WITH_SPNG "Include SPNG support" OFF`. So we can add libspng as optional codec, not default as we discussed in one of the GSoC meetings with the Vadim and Suleyman.